src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -631,27 +631,20 @@ IEnumerable<DataViewSchema.Column> ISchemaBoundRowMapper.GetDependenciesForNewCo
                 return _predictor.GetInputColumnRoles();
             }
 
-            public DataViewRow GetRow(DataViewRow input, Func<int, bool> predicate)
+            DataViewRow ISchemaBoundRowMapper.GetRow(DataViewRow input, IEnumerable<DataViewSchema.Column> activeColumns)
             {
-                Func<int, bool> predictorPredicate = col => false;
-                for (int i = 0; i < OutputSchema.Count; i++)
-                {
-                    if (predicate(i))
-                    {
-                        predictorPredicate = col => true;
-                        break;
-                    }
-                }
-                var predictorRow = _predictor.GetRow(input, predictorPredicate);
+                var predictorRow = _predictor.GetRow(input, activeColumns.Count() > 0 ? OutputSchema : Enumerable.Empty<DataViewSchema.Column>());
                 var getters = new Delegate[OutputSchema.Count];
-                for (int i = 0; i < OutputSchema.Count - 1; i++)
+
+                foreach (var column in activeColumns)
                 {
-                    var type = predictorRow.Schema[i].Type;
-                    if (!predicate(i))
+                    if (column.Index == OutputSchema.Count - 1)
                         continue;
-                    getters[i] = Utils.MarshalInvoke(GetPredictorGetter<int>, type.RawType, predictorRow, i);
+                    var type = predictorRow.Schema[column.Index].Type;
+                    getters[column.Index] = Utils.MarshalInvoke(GetPredictorGetter<int>, type.RawType, predictorRow, column.Index);
                 }
-                if (predicate(OutputSchema.Count - 1))
+
+                if (activeColumns.Select(c => c.Name).Contains(DefaultColumnNames.Probability))
```

Comment:
> if (activeColumns.Select(c => c.Name).Contains(DefaultColumnNames.Probability)) [](start = 15, length = 80)

I would rather change if inside foreach loop to set bool flag to true if column.Index == OutputSchema.Count-1 instead of continue and check that flag here.
You have mix of two different conditions right now.
And who knows maybe we will allow user to define their own names for probability column in future, so index feels somewhat more secure.